examples/ip_adapter/tools/merge_ckpts_sdxl.py

```diff
@@ -1,13 +1,90 @@
+    content = replace(content, "input_blocks.3.attentions.1", "input_blocks.11.1")
+    content = replace(content, "middle_block.resnets.0", "middle_block.0")
+    content = replace(content, "middle_block.attentions.0", "middle_block.1")
+    content = replace(content, "middle_block.resnets.1", "middle_block.2")
+
+    content = replace(content, "conv1", "in_layers.0")
+    content = replace(content, "conv2", "out_layers.0")
+    content = replace(content, "time_emb_proj", "emb_layers.1")
+    content = replace(content, "conv_shortcut", "skip_connection")
+    content = replace(content, "controlnet_middle_block", "middle_block_out.0")
+
+    content = replace(content, "input_hint_block.conv_in", "input_hint_block.0")
+    content = replace(content, "input_hint_block.blocks.0", "input_hint_block.2")
+    content = replace(content, "input_hint_block.blocks.1", "input_hint_block.4")
+    content = replace(content, "input_hint_block.blocks.2", "input_hint_block.6")
+    content = replace(content, "input_hint_block.blocks.3", "input_hint_block.8")
+    content = replace(content, "input_hint_block.blocks.4", "input_hint_block.10")
+    content = replace(content, "input_hint_block.blocks.5", "input_hint_block.12")
+    content = replace(content, "input_hint_block.conv_out", "input_hint_block.14")
+
+    content = re.sub(r"(zero_convs.[0-9]*).weight", r"\1.0.weight", content)
+    content = re.sub(r"(zero_convs.[0-9]*).bias", r"\1.0.bias", content)
+    content = re.sub(r"(norm[0-9]*).weight", r"\1.gamma", content)
+    content = re.sub(r"(norm[0-9]*).bias", r"\1.beta", content)
+
+    content = replace(content, "in_layers.0.bias", "in_layers.2.bias")
+    content = replace(content, "in_layers.0.weight", "in_layers.2.weight")
+    content = replace(content, "out_layers.0.bias", "out_layers.3.bias")
+    content = replace(content, "out_layers.0.weight", "out_layers.3.weight")
+
+    content = re.sub(r"(input_blocks.[0-9]*.0).norm1.beta", r"\1.in_layers.0.beta", content)
+    content = re.sub(r"(input_blocks.[0-9]*.0).norm1.gamma", r"\1.in_layers.0.gamma", content)
+    content = re.sub(r"(input_blocks.[0-9]*.0).norm2.beta", r"\1.out_layers.0.beta", content)
+    content = re.sub(r"(input_blocks.[0-9]*.0).norm2.gamma", r"\1.out_layers.0.gamma", content)
+    content = re.sub(r"(middle_block.[0-9]*).norm1.beta", r"\1.in_layers.0.beta", content)
+    content = re.sub(r"(middle_block.[0-9]*).norm1.gamma", r"\1.in_layers.0.gamma", content)
+    content = re.sub(r"(middle_block.[0-9]*).norm2.beta", r"\1.out_layers.0.beta", content)
+    content = re.sub(r"(middle_block.[0-9]*).norm2.gamma", r"\1.out_layers.0.gamma", content)
+
+    content = replace(content, "conv.bias", "op.bias")
+    content = replace(content, "conv.weight", "op.weight")
+
+    content = replace(content, "add_embedding.linear_1", "label_emb.0.0")
+    content = replace(content, "add_embedding.linear_2", "label_emb.0.2")
```

Comment:
Better to define a name mapping dict and use for-loop to do the replacement.

Reply:
I agree. It is a temporary converting script for this project only.